Sources/Query.swift

```diff
@@ -8,159 +8,507 @@
 
 import Foundation
 import Interstellar
+import CoreLocation
 
+/// The available URL parameter names for queries; used internally by the various `Contentful.Query` types.
+/// If you are constructing Dictionary's to pass into `Client` "fetch" methods, instead of taking advantage of "fetch" methods
+/// that take `Query` types, use these static variables to avoid typos in query construction.
 public struct QueryParameter {
 
-    static let contentType  = "content_type"
-    static let select       = "select"
-    static let order        = "order"
-    static let limit        = "limit"
-    static let skip         = "skip"
-    static let include      = "include"
-    static let locale       = "locale"
+    static let contentType      = "content_type"
+    static let select           = "select"
+    static let order            = "order"
+    static let limit            = "limit"
+    static let skip             = "skip"
+    static let include          = "include"
+    static let locale           = "locale"
+    static let mimetypeGroup    = "mimetype_group"
+    static let fullTextSearch   = "query"
 }
 
+fileprivate struct QueryConstants {
+    static let maxLimit: UInt               = 1000
+    static let maxSelectedProperties: UInt  = 99
+}
+
+/// Use types that conform to QueryableRange to perform queries with the four Range operators
+/// See: <https://www.contentful.com/developers/docs/references/content-delivery-api/#/reference/search-parameters/ranges>
+public protocol QueryableRange {
+
+    var stringValue: String { get }
+}
+
+extension Int: QueryableRange {
+
+    public var stringValue: String {
+        return String(self)
+    }
+}
+
+extension String: QueryableRange {
+    public var stringValue: String {
+        return self
+    }
+}
+
+extension Date: QueryableRange {
+
+    public var stringValue: String {
+        return self.toISO8601GMTString()!
+    }
+}
 
+/// Use bounding boxes or bounding circles to perform queries on location-enabled content.
```

Comment:
Another stupid question: Is there a significant difference between using `/** **/` and `///` ? If not, I'd run for consistency and decided for `/**` in type comments.